src/command/defaults/WeatherCommand.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\command\defaults;
+
+use pocketmine\command\CommandSender;
+use pocketmine\lang\Translatable;
+use pocketmine\permission\DefaultPermissionNames;
+use pocketmine\player\Player;
+use pocketmine\world\WeatherType;
+use function count;
+use function max;
+use function strtolower;
+
+class WeatherCommand extends VanillaCommand{
+
+	public function __construct(){
+		parent::__construct(
+			"weather",
+			new Translatable("commands.weather.description"),
+			"/weather <clear|rain|thunder> [duration]"
+		);
+		$this->setPermission(DefaultPermissionNames::COMMAND_WEATHER);
+	}
+
+	public function execute(CommandSender $sender, string $label, array $args) : void{
+		if(!$sender instanceof Player){
+			$sender->sendMessage(new Translatable("commands.generic.ingame"));
+			return;
+		}
+
+		if(!$this->testPermission($sender)){
+			return;
+		}
+
+		$world = $sender->getWorld();
+		$weatherManager = $world->getWeatherManager();
+
+		if(count($args) < 1){
+			/** @var WeatherType $current */
+			$current = $weatherManager->getWeather();
+			$stateName = match($current){
+				WeatherType::CLEAR => new Translatable("commands.weather.query.clear"),
+				WeatherType::RAIN => new Translatable("commands.weather.query.rain"),
+				WeatherType::THUNDER => new Translatable("commands.weather.query.thunder"),
+			};
+
+			$sender->sendMessage(new Translatable("commands.weather.query", [$stateName]));
+			return;
+		}
+
+		$type = match(strtolower($args[0])){
+			"clear" => WeatherType::CLEAR,
+			"rain" => WeatherType::RAIN,
+			"thunder" => WeatherType::THUNDER,
+			default => null
+		};
+
+		if($type === null){
+			$sender->sendMessage("Invalid weather type. Use: clear, rain, or thunder.");
```

Comment:
This should throw `InvalidCommandSyntaxException` instead of sending a hardcoded message